rotkehlchen/exchanges/binance.py

```diff
@@ -1205,15 +1407,29 @@ def _api_query_list_within_time_delta(
             else end_ts  # Case request without offset (1 request)
         )
         while True:
+            current_page = 1
             options = {
                 'timestamp': ts_now_in_ms(),
                 'startTime': from_ts,
                 'endTime': to_ts,
             }
             if additional_options:
                 options.update(additional_options)
-            result = self.api_query_list(api_type, method, options=options)
-            results.extend(result)
+                if method == 'lending/union/interestHistory':
+                    options['current'] = current_page
+
+            if method == 'lending/union/interestHistory':
+                while True:
```

Comment:
This is dangerous. We should not have an unprotected while True in the code. There should be some kidn of limit and auto-breaking.

Reply:
There's auto-breaking two lines below where it checks if the result is an empty list or not

Reply:
This is not auto breaking dude wtf? This depends on external input. If somehow the binance api is returning us the same page for whatever reason this is an infinite loop.